zerver/lib/actions.py

```diff
@@ -2761,6 +2788,27 @@ def our_person(p):
                         bot.update(event['bot'])
 
         elif event['type'] == 'stream':
+            if event['op'] == 'create':
+                for stream in event['streams']:
+                    if not stream['invite_only']:
+                        stream_name = stream['name']
+                        realm = get_realm(user_profile.realm.domain)
+                        new_stream = get_stream(stream_name, realm)
+                        subscribers = get_subscriber_ids(new_stream)
+                        stream_data = {
+                            'description': new_stream.description.decode('utf-8'),
+                            'stream_id': new_stream.id,
+                            'subscribers':  subscribers,
+                            'invite_only': new_stream.invite_only,
+                            'name': new_stream.name.decode('utf-8')
+                        }
```

Comment:
I think you can replace the block inside the `invite_only` if statement with:
```
stream_data = copy.deepcopy(stream)
stream_data['subscribers'] = []
state['never_subscribed'].append(stream_data)
```
since structurally, a newly created stream never has any subscribers (even if they are added in the same API call that created the stream, there would be a later event in the series of events that would add the subscribers to the new stream).